### PR TITLE
Update

### PR DESCRIPTION
**Pull request by karantrehan3**
Removed Callbacks, Used async-await everywhere.
Brought the Validation Middleware in the code.
Added two fields in the Note, 'createdAt' and 'lastModifiedAt'
Enhanced the Folder Structure of the Project by introducing relevant Folders.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -1,81 +1,138 @@
...
+router.get("/list", async (req, res) => {
+    notes
+        .loadNotes()
+        .then((data) => {
+            res.status(200).send(data);
+        })
+        .catch((err) => {
+            res.status(500).send(err);
+            console.log(err);
+        });
```
Comment (avram19) [inline routes/index.js]: you can use async await with try/catch to do the same thing. Would be more easier to read.

File: src/notes.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                body: body,
+                createdAt: timestamp,
+                lastModifiedAt: timestamp,
+            });
+            await saveNotes(notes);
+            return true;
+        }
+        return false;
+    } catch (err) {
+        console.error(err);
```
Comment (avram19) [inline src/notes.js]: Return Promise reject in catch block, so that the calling function can decide what can be done.

File: routes/index.js (hunk 1 above)
Comment (karantrehan3) [reply routes/index.js]: Fixed this. Kindly review!

File: src/notes.js (hunk 2 above)
Comment (karantrehan3) [reply src/notes.js]: Fixed this. Kindly review!

**Pull request merged**